### PR TITLE
lkl: avoid dhcp client running if ipv6 addr is specified

### PR DESCRIPTION
**Pull request by thehajime**
Fixes: 817b3acc6e2e ("lkl: add dhcp client option")
Signed-off-by: Hajime Tazaki <thehajime@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/lkl/linux/278)

### CODE REVIEW

**Comment by liuyuan10** [comment]
Can't dhcp co-exist with ipv6?

**Comment by thehajime** [comment]
> Can't dhcp co-exist with ipv6?

thanks for the comment.  it should co-exist indeed.

817b3ac introduced that dhcp is a default option when LKL_HIJACK_NET_IP is not specified, but it makes IPv6 only configs annoying since a program will wait until DHCP timeout.

I think I will revert this behavior to disabling dhcp by default, and initiate dhcp client only when LKL_HIJACK_NET_IP is specified with `dhcp`.

File: tools/lkl/lib/hijack/init.c [hunk 1]
```diff
@@ -325,9 +324,9 @@ hijack_init(void)
 	if (single_cpu_mode == 1)
 		PinToFirstCpu(&ori_cpu);
 
-	/* clear cmdline if non-dhcp case */
-	if ((ip && strcmp(ip, "dhcp")) || (nd_id == -1))
-		boot_cmdline = "";
+	/* set cmdline if dhcp case */
+	if ((ip && !strcmp(ip, "dhcp")) || (nd_id != -1))
```
Comment (liuyuan10) [inline tools/lkl/lib/hijack/init.c]: Is it should be "&&" (nd_id != -1)?

Comment (thehajime) [inline tools/lkl/lib/hijack/init.c]: gee, you're right. should be fixed now.

**Comment by tavip** [comment]
Can one of the admins verify this patch?

**Pull request merged**